src/pages/billing/Project.js

```diff
@@ -148,31 +147,26 @@ const ProjectDetail = ({ project, project: { projectName }, billingAccounts, aut
   const [expandedWorkspaceName, setExpandedWorkspaceName] = useState()
   const [sort, setSort] = useState({ field: 'email', direction: 'asc' })
   const [workspaceSort, setWorkspaceSort] = useState({ field: 'name', direction: 'asc' })
+  const [billingProject, setBillingProject] = useState(project)
 
   const signal = Utils.useCancellation()
 
   const adminCanEdit = _.filter(({ roles }) => _.includes(billingRoles.owner, roles), projectUsers).length > 1
 
-  const workspacesInProject = useMemo(() => _.flow(
-    _.map(({ workspace }) => workspace),
-    _.filter({ namespace: projectName })
-  )(workspaces), [workspaces, projectName])
-
-  const getBillingAccountStatusIcon = (() => {
-    const allWorkspacesHaveCorrectBillingAccount =
-      _.every({ billingAccount: billingAccountName }, workspacesInProject)
-    return workspace => Utils.cond(
-      [allWorkspacesHaveCorrectBillingAccount, () => null],
-      [billingAccountName === workspace.billingAccount, () => 'success-standard'],
-      ['billingAccountErrorMessage' in workspace, () => 'error-standard'],
-      [Utils.DEFAULT, () => 'loadingSpinner'])
-  })()
+  const getBillingAccountStatusIcon = workspace => Utils.cond(
+    [_.isEmpty(billingProject.workspacesWithIncorrectBillingAccount), () => null],
+    [billingProject.billingAccount === workspace.billingAccount, () => 'success-standard'],
+    ['billingAccountErrorMessage' in workspace, () => 'error-standard'],
```

Comment:
To avoid the surprises of `in`:
```suggestion
    [_.has(workspace, 'billingAccountErrorMessage'), () => 'error-standard'],
```

Reply:
`has` exhibits the same behaviour as `in` for your example: 
```javascript
> 0 in [1, 2, 3] === _.has(0, [1, 2, 3])
true
```